src/tigerbeetle.zig

```diff
@@ -32,6 +32,19 @@ pub const Account = extern struct {
         return (self.flags.credits_must_not_exceed_debits and
             self.credits_pending + self.credits_posted + amount > self.debits_posted);
     }
+
+    pub fn equal(self: *const Account, other: *const Account) bool {
```

Comment:
Would it make more sense to compare with `mem.eql(mem.asBytes(), mem.asBytes())`? (Also in `AccountFlags`). That way it is obvious nothing is missed (e.g. padding).

Reply:
Is the lsm required to preserve padding?

Reply:
Yes; padding for `Account`/`Transfer` is explicit (and checked). e.g. we assert `assert(@bitSizeOf(Account) == @sizeOf(Account) * 8);` — no implicit padding.

- https://github.com/tigerbeetledb/tigerbeetle/blob/339200dd1bc78ae191dbcb0df6fe44336658eceb/src/tigerbeetle.zig#L51
- https://github.com/tigerbeetledb/tigerbeetle/blob/339200dd1bc78ae191dbcb0df6fe44336658eceb/src/tigerbeetle.zig#L10
- https://github.com/tigerbeetledb/tigerbeetle/blob/339200dd1bc78ae191dbcb0df6fe44336658eceb/src/state_machine.zig#L532-L533

Reply:
:+1: